avocado/core/runner.py

```diff
@@ -234,21 +229,13 @@ def _run_test(self, test_factory, queue):
         sys.stdout = output.LoggingFile(logger=logger_list_stdout)
         sys.stderr = output.LoggingFile(logger=logger_list_stderr)
 
-        try:
-            instance = loader.load_test(test_factory)
-            if instance.runner_queue is None:
-                instance.runner_queue = queue
-            runtime.CURRENT_TEST = instance
-            early_state = instance.get_state()
-            early_state['early_status'] = True
-            queue.put(early_state)
-        except Exception:
-            exc_info = sys.exc_info()
-            app_logger = logging.getLogger('avocado.app')
-            app_logger.exception('Exception loading test')
-            tb_info = stacktrace.tb_info(exc_info)
-            queue.put({'load_exception': tb_info})
-            return
+        instance = loader.load_test(test_factory)
+        if instance.runner_queue is None:
+            instance.runner_queue = queue
+        runtime.CURRENT_TEST = instance
+        early_state = instance.get_state()
+        early_state['early_status'] = True
+        queue.put(early_state)
```

Comment:
Actually if there is an exception before early_state, the process dies and we get an immediate exception:

```
JOB ID     : 662f0d352e81c8575558c74e6f23c00253d05727
JOB LOG    : /home/medic/avocado/job-results/job-2016-04-13T12.08-662f0d3/job.log
TESTS      : 1

Avocado crashed: TestError: Process died before it pushed early test_status.
Traceback (most recent call last):

  File "/home/medic/Work/Projekty/avocado/avocado/avocado/core/job.py", line 502, in _run
    replay_map=replay_map)

  File "/home/medic/Work/Projekty/avocado/avocado/avocado/core/runner.py", line 451, in run_suite
    summary, deadline)

  File "/home/medic/Work/Projekty/avocado/avocado/avocado/core/runner.py", line 315, in run_test
    test_status.wait_for_early_status(proc, 10)

  File "/home/medic/Work/Projekty/avocado/avocado/avocado/core/runner.py", line 114, in wait_for_early_status
    raise exceptions.TestError("Process died before it pushed "

TestError: Process died before it pushed early test_status.

Please include the traceback info and command line used on your bug report
Report bugs visiting https://github.com/avocado-framework/avocado/issues/new
```

So not even the 10s timeout. I'm wondering if that's the best we can do, but IMO it's a huge corner case and failing the whole avocado is appropriate response.